src/ProgrammerGui/ProgrammerGuiIntegration.cpp

```diff
@@ -78,31 +78,40 @@ void ProgrammerGuiIntegration::Devices(const Cable &cable,
 }
 
 void ProgrammerGuiIntegration::Progress(const std::string &progress) {
-  emit this->progress(progress);
+  emit this->progress({m_current.first, m_current.second, m_type}, progress);
 }
 
 void ProgrammerGuiIntegration::ProgramFpga(const Cable &cable,
                                            const Device &device,
                                            const std::string &file) {
   m_current = std::make_pair(cable, device);
-  m_flash = false;
+  m_type = Fpga;
```

Comment:
To be explicit and also consistent to line 105
```suggestion
  m_type = Type::Fpga;
```

Reply:
done